src/npc_attack.cpp

```diff
@@ -329,10 +399,13 @@ void npc_attack_activate_item::use( npc &source, const tripoint &/*location*/ )
 
 bool npc_attack_activate_item::can_use( const npc &source ) const
 {
-    const bool can_use_grenades = ( source.is_player_ally() &&
-                                    !source.rules.has_flag( ally_rule::use_grenades ) ) || source.is_hallucination();
-    //TODO: make this more complete. only does BOMB type items
-    return can_use_grenades && activatable_item.has_flag( flag_NPC_ACTIVATE );
+    if( !activatable_item.has_flag( flag_NPC_ACTIVATE ) ) {
```

Comment:
i don't understand the purpose of churning this function. your changes are the exact same result.

Reply:
Bailing early for items that aren't activable so more checks (is_player_ally(), etc) don't have to be done. If more checks are added, this will still cull any extra checks for those items.